### DISCUSSION TITLE
Secure Session Cookies

### DISCUSSION THREAD

**Discussion by mike-stewart** [open-webui/open-webui#4233] [General]
# Bug Report

## Description

**Bug Summary:**
The WEBUI_SESSION_COOKIE_SECURE environment variable sets the secure flag on the "oui-session" cookie, but not the "token" cookie.

Here it is set: https://github.com/open-webui/open-webui/blob/82079e644a291349926008adffe8fe2fc3b33566/backend/main.py#L2118-L2124

Here it is not set:
https://github.com/open-webui/open-webui/blob/82079e644a291349926008adffe8fe2fc3b33566/backend/main.py#L2243-L2247
https://github.com/open-webui/open-webui/blob/82079e644a291349926008adffe8fe2fc3b33566/backend/apps/webui/routers/auths.py#L59-L63

**Steps to Reproduce:**
Login with oauth.

**Expected Behavior:**
All sensitive cookies should have the secure flag set when `WEBUI_SESSION_COOKIE_SECURE=true`.

**Actual Behavior:**
The `token` cookie does not have the Secure flag set.

## Environment

- **Open WebUI Version:** v0.3.10

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
N/A

**Docker Container Logs:**
N/A

**Screenshots (if applicable):**
![Screenshot 2024-07-30 at 9 01 47 PM](https://github.com/user-attachments/assets/367c9401-3d71-4237-af80-640f45925519)

## Installation Method

Docker

## Additional Information

N/A

**Comment by tjbck** [comment]
PR Welcome

**Reply by mike-stewart** [reply]
@tjbck Should we use the same WEBUI_SESSION_COOKIE_SAME_SITE and WEBUI_SESSION_COOKIE_SECURE in both places?

**Comment by Zeeeeta** [comment]
WEBUI_SESSION_COOKIE_SECURE is very weird, I had no problem setting it to True in docker compose and running in container on couple of environments:
- Docker for Mac, listening to http 3000, behind a load balancer/proxy with a intranet domain and https, accessed from company PCs
- Docker for Windows, listening to http 3000, accessed locally with localhost

HOWEVER, it has issue no matter setting to True or False when running on a Linux Docker, listening to http 3000, accessed directly from company PCs with a hostname registered on DNS.
I had to remove this env var completely to have it works normally.
The following is the personal remark I wrote for myself after finding out the fix/workaround:

```
environment:
#  WEBUI_SESSION_COOKIE_SECURE will for some reason cause problem in certain environment (e.g. http Open WebUI running on Linux docker),
#  and lead to issue such as CSRF state mismatch for SSO, and page hang when going to Settings > Audio or Admin Settings > Interface,
#  no matter if it is set to True or False.
#  - 'WEBUI_SESSION_COOKIE_SECURE=True'
```
